File: src/1214-Two-Sum-BSTs/1214.go
func twoSumBSTs(root1 *TreeNode, root2 *TreeNode, target int) bool {
    if root1 == nil {
        return false
    }
    if find(root2, target - root1.Val) {
        return true
    }
    return twoSumBSTs(root1.Left, root2, target) || twoSumBSTs(root1.Right, root2, target)
}

func find(root *TreeNode, target int) bool {
    if root == nil {
        return false
    }
    if target == root.Val {
        return true
    } else if target < root.Val {
        return find(root.Left, target)
    } else {
        return find(root.Right, target)  
    }
}

File: src/1214-Two-Sum-BSTs/1214.py
class Solution:
    def twoSumBSTs(self, root1: TreeNode, root2: TreeNode, target: int) -> bool:
        def find(root, t):
            if not root:
                return False
            if root.val == t:
                return True
            elif root.val < t:
                return find(root.right, t)
            else:
                return find(root.left, t)
        
        if not root1:
            return False
        if find(root2, target - root1.val):
            return True
        return self.twoSumBSTs(root1.left, root2, target) or \
            self.twoSumBSTs(root1.right, root2, target)